xmake/modules/core/tools/cl.lua

```diff
@@ -136,7 +136,7 @@ function nf_warning(self, level)
         none       = "-W0"
     ,   less       = "-W1"
     ,   more       = "-W3"
-    ,   all        = "-W3" -- = "-Wall" will enable too more warnings
+    ,   all        = "-W4" -- = "-Wall" will enable too more warnings
```

Comment:
Here too, this will cause more warnings to be triggered, affecting existing projects

Reply:
I agree this is a breaking change, I leave you the choice on whether it's too much of a change or not (I'll just argue that there's not so many additional warnings, and that simply upgrading a compiler also enables more warnings).

If you decide it's too much, maybe provide an "extra" warning level, enabling `/W4` and `-Wall -Wextra`?

Reply:
We can add a warning level `extra` to correspond to `-Wextra`

`set_warnings("all", "extra", "error")`

Reply:
Sure, do you want me to try to do it on this PR?

Reply:
Yes, Thanks.